src/plasmapy/simulation/particle_tracker/particle_tracker.py

```diff
@@ -429,12 +453,99 @@ def load_particles(
         self.x = x.to(u.m).value
         self.v = v.to(u.m / u.s).value
 
+    def add_Bethe_stopping(self, I: u.Quantity[u.J]):  # noqa: E741
+        r"""
+        Enable particle stopping described by non-relativistic Bethe formula.
+
+        The electron density associated with each material must be provided in
+        the grid during instantiation.
+
+        Parameters
+        ----------
+        I : `~astropy.units.Quantity`, shape (ngrids)
+            An array of the mean excitation energy values for each of the provided
+            grids.
+
+        Raises
+        ------
+        `ValueError`:
+            The provided mean excitation energy array does not match with the number
+            of grids provided at class instantiation.
+        """
+
+        if len(I) != len(self.grids):
+            raise ValueError(
+                "Please provide an array of length ngrids for the mean excitation energies."
+            )
+
+        self._do_stopping = True
+        self._stopping_routine = "Bethe stopping"
+
+        self._excitation_energies = I.to(u.J).value
+
+        # Require that each grid has a defined electron density,
+        # if a grid does not define n_e, raise an exception
+        for grid in self.grids:
+            grid.require_quantities("n_e", replace_with_zeros=False)
+
+    @staticmethod
+    def _is_quantity_defined_on_one_grid(
+        grids: list[AbstractGrid], quantity: str
+    ) -> bool:
+        r"""
+        Check to ensure the provided quantity string is defined on at least one grid.
+
+        Returns ``True`` if the quantity is defined on at least one grid and
+        ``False`` if none of the grids have defined the specified quantity.
+        """
+
+        return any(quantity in grid.quantities for grid in grids)
+
+    def add_stopping(self, materials: list[str]):
```

Comment:
What if we handled this like the termination condition and saving routines? That way the user can define their own stopping if they wanted

Reply:
Not really a stopping routine though, just a model for the stopping power? In principle I agree this would be nice, but in practice I think this adds complexity and isn't maybe necessary yet - if people start using this a lot and that becomes a problem, we could split it off like that then?